src/OptimalTransport.jl

```diff
@@ -715,33 +715,82 @@ Returns the entropically regularised barycenter of the `μ`, i.e. the histogram
 where ``\\operatorname{OT}_{ε}(\\mu, \\nu) = \\inf_{\\gamma \\Pi(\\mu, \\nu)} \\langle \\gamma, C \\rangle + \\varepsilon \\Omega(\\gamma)`` 
 is the entropic optimal transport loss with cost ``C`` and regularisation ``\\epsilon``.
 """
-function sinkhorn_barycenter(μ, C, ε, w; tol=1e-9, check_marginal_step=10, max_iter=1000)
-    sums = sum(μ; dims=1)
-    if !isapprox(extrema(sums)...)
-        throw(ArgumentError("Error: marginals are unbalanced"))
+function sinkhorn_barycenter(
+    μ,
+    C,
+    ε,
+    w;
+    tol=nothing,
+    atol=tol,
+    rtol=nothing,
+    check_marginal_step=nothing,
+    check_convergence=10,
+    max_iter=nothing,
+    maxiter=1_000,
+)
+    if tol !== nothing
+        Base.depwarn(
+            "keyword argument `tol` is deprecated, please use `atol` and `rtol`",
+            :sinkhorn_barycenter,
+        )
+    end
+    if check_marginal_step !== nothing
+        Base.depwarn(
+            "keyword argument `check_marginal_step` is deprecated, please use `check_convergence`",
+            :sinkhorn_barycenter,
+        )
     end
+    if max_iter !== nothing
+        Base.depwarn(
+            "keyword argument `max_iter` is deprecated, please use `maxiter`",
+            :sinkhorn_barycenter,
+        )
+    end
+    if !isapprox(extrema(sum(μ; dims=1))...)
+        throw(ArgumentError("Error: input marginals must have the same mass"))
+    end
+
+    T = float(Base.promote_eltype(μ, C))
+    _atol = atol === nothing ? 0 : atol
+    _rtol = rtol === nothing ? (_atol > zero(_atol) ? zero(T) : sqrt(eps(T))) : rtol
+
     K = exp.(-C / ε)
-    converged = false
-    v = ones(size(μ))
-    u = ones(size(μ))
+    v = similar(μ, T)
+    u = similar(μ, T)
+    fill!(u, one(T))
+    a = similar(u, T, size(u, 1), 1)
+    tmp = similar(u)
     N = size(μ, 2)
-    for n in 1:max_iter
-        v = μ ./ (K' * u)
-        a = ones(size(u, 1))
-        a = prod((K * v)' .^ w; dims=1)'
-        u = a ./ (K * v)
-        if n % check_marginal_step == 0
+    # norm of inputs for convergence check
+    norm_μ = sum(abs, μ; dims=1)
+    isconverged = false
+    for iter in 1:maxiter
+        mul!(tmp, K', u)
+        @. v = μ / tmp
+        mul!(tmp, K, v)
+        a .= prod(tmp' .^ w; dims=1)'
+        @. u = a / tmp
```

Comment:
This could be simplified to
```suggestion
        mul!(tmp, K, v)
        tmp2 .= tmp .^ w'
        prod!(a, tmp2)
        @. u = a / tmp
```
where `tmp2` is an array of appropriate size.